tmol/tests/score/common/test_energy_term.py

```diff
@@ -0,0 +1,428 @@
+
+        coords = torch.nn.Parameter(p1.coords.clone())
+        scores = (
+            pose_scorer(coords, output_block_pair_energies=True).cpu().detach().numpy()
+        )
+
+        if update_baseline:
+            cls.save_test_baseline_data(
+                cls.test_block_scoring.__name__, cls.block_pair_to_dict(scores)
+            )
+        gold_vals = cls.get_test_baseline_data(cls.test_block_scoring.__name__)
+
+        assert_allclose(gold_vals, scores, atol, rtol)
+
+    @classmethod
+    def test_block_scoring_reweighted_gradcheck(
+        cls,
+        pdb,
+        default_database,
+        torch_device,
+        resnums=None,
+        eps=1e-6,  # torch default
+        atol=1e-5,  # torch default
+        rtol=1e-3,  # torch default
+        nondet_tol=0.0,  # torch default
+    ):
+        p1 = pose_stack_from_pdb_and_resnums(pdb, torch_device, resnums)
+
+        pose_scorer = cls.get_pose_scorer(p1, default_database, torch_device)
+
+        def score(coords):
+            scores = pose_scorer(coords, output_block_pair_energies=True)
+            scale = 0.01 * torch.arange(
+                torch.numel(scores), device=scores.device
+            ).reshape(scores.shape)
+            return torch.sum(scale * scores)
+
+        # monkeypatch more sane error reporting
+        torchgrad = importlib.import_module("torch.autograd.gradcheck")
+        torchgrad._get_notallclose_msg = functools.partial(
+            _get_notallclose_msg, atol=atol, rtol=rtol
+        )
+
+        torchgrad.gradcheck(
+            score,
+            (p1.coords.double().requires_grad_(True),),
+            eps=eps,
+            atol=atol,
+            rtol=rtol,
+            nondet_tol=nondet_tol,
+        )
+
+
+# We use Ref as dummy for generic energy term tests. This new class ensures it gets its own test data directory
+class DummyEnergyTerm(RefEnergyTerm):
+    pass
+
+
+class EnergyTermBaseTester(EnergyTermTestBase):
+    energy_term_class = DummyEnergyTerm
+
+
+# This test just makes sure the updating_baseline functionality works (only tests the '10' variant currently)
+def test_energy_term_base_write_baseline_smoke(ubq_pdb, default_database, torch_device):
+    test_class = EnergyTermBaseTester()
+    test_class.test_whole_pose_scoring_10(
+        ubq_pdb, default_database, torch_device, update_baseline=True
+    )
+
+
+# This test makes sure that the 'jagged' test of the DummyEnergyTerm (Ref) fails. The baselines for the Dummy term's 'jagged' test have been manually modified and should always fail
+def test_energy_term_fail(ubq_pdb, default_database, torch_device):
+    test_class = EnergyTermBaseTester()
+    failed = False
+
+    try:
+        test_class.test_whole_pose_scoring_jagged(
+            ubq_pdb, default_database, torch_device, update_baseline=False
+        )
+        failed = True
+    except AssertionError:
+        pass
+
+    if failed:
+        raise AssertionError("Test passed with bad baselines.")
```

Comment:
Added these tests to for the term testing machinery (and to make codecov pass). Let me know if this looks good or we want anything additional.